src/main/webapp/js/lib/util.js

```diff
@@ -55,6 +56,21 @@ function createPlanTable(tableData, creditsData, courseContainer) {
   courseContainer.appendChild(table);
 }
 
+/**
+ * Rectangularizes matrix by ensuring each row has an equal number of elements.
+ * @param {Object} data The 2d array to rectangularize
+ */
+function rectangularize2dMatrix(data) {
+  const longestRow =
```

Comment:
why can't this be:

`const longestRow = Math.max(data.map(arr => arr.length));`

Reply:
I tried that first, it results in NaN for some reason.

Reply:
Ah, you need: `Math.max(...data.map(arr => arr.length));`

Math.max takes in a variable number of arguments, not an array.

Reply:
that works!